HeartBeat.php

```diff
@@ -75,31 +70,52 @@ public function newDirectRequest()
      */
     public function getPluginSettings($getValues=true)
     {
-        /* We fix the default before read the parent: the arent set the curretnt to actual default */
-        $this->settings['interval']['default']=self::getDefaultInterval();
-        $this->settings['interval']['help']="Your actual session.gc_maxlifetime is ".App()->session->getTimeout()."s";
+        /* We fix the default before read the parent: the parent set the current to actual default */
+        $this->settings['interval']['default'] = self::getDefaultInterval();
+        $this->settings['useSessionLifeTime']['help'] = sprintf($this->_translate("Your actual session.gc_maxlifetime is %s seconds"),App()->session->getTimeout());
+        $this->settings['interval']['label'] = $this->_translate("Interval of the heartbeat (in seconds, minimum: 5s)");
         $aPluginSettings = parent::getPluginSettings($getValues);
 
         $useSessionLifeTime = (boolean) $this->get('useSessionLifeTime', null, null, false);
-
-        // If it is a post request, it could be an autosubmit so read posted
-        // value over the saved value
-        if (App()->request->isPostRequest)
-        {
+        if (App()->request->isPostRequest) {
             $useSessionLifeTime = App()->request->getPost('useSessionLifeTime', $useSessionLifeTime);
+            if(empty($useSessionLifeTime) && intval(App()->request->getPost('interval')) < 5) {
+                if(empty(App()->request->getPost('interval'))) {
```

Comment:
`intval(App()->request->getPost('interval')) < 5` can be removed and replaced with checking if it's empty.

Reply:
Arg … in fact it don't change anything but yes …